src/export/StructureDefinitionExporter.ts

```diff
@@ -414,9 +414,9 @@ export class StructureDefinitionExporter implements Fishable {
         // The FHIR spec prohibits constraining any parent element in a 'specialization'
         // (i.e., logical model and resource), therefore log an error if that is attempted
         // and continue to the next rule.
-        if (element.path !== element.base.path) {
-          // The AddElementRule always sets the newElementBase.path to the value of element.path.
-          // All parent elements will have the element.base.path pointing to the parent.
+        const baseElementPath = element.path.replace(`${element.structDef.id}.`, '');
```

Comment:
I think that technically the first part of the path is the `element.structDef.type`.  That will be the same as the `element.structDef.id` in most (or all?) cases for logical models and resources -- but I still think it's more correct to use `type` here rather than `id`.